msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -110,6 +110,37 @@ To initiate actions from a  messaging extension set the `type` parameter to `act
 }
 ```
 
+### Initiate actions from messages
+
+> [!NOTE]
+> Initiating actions from messages is in [developer preview](~/resources/dev-preview/developer-preview-intro.md).
+
+In addition to initiating actions from the compose message area, you can also use your messaging extension to initiate an action from a message. This will allow you to send the contents of the message to your bot for processing, and optionally reply to that message with the results. Your users can access your messaging extension from the overflow `...` menu and then selecting `Take action` as in the image below.
+
+![Example of initiating an action from a message](~/assets/images/compose-extensions/messageextensions_messageaction.png)
+
+To enable your messaging extension to work from an action you'll need to add the `context` parameter to your messaging extension's `commands` object in your app manifest as in the example below. Valid strings for the `context` array are `"message"`, `"commandBar"`, and `"compose"`. The default value is `["compose", "commandBar"]`.
+
+```json
+"composeExtensions": [
+  {
+    "botId": "12345ab8-ab12-cd34-ef56-098abc123876",
+    "canUpdateConfiguration": true,
+    "commands": [
+      {
+        "id": "reassignTodo",
+        "description": "Reassign a todo item",
+        "title": "Create To Do",
+        "type": "Action",
+        "context": ["message"],
+        "fetchTask": true
+    }]
+    ...
+
+```
+
+The request sent to your bot is the same as described [below](#responding-to-submit), where the `content` object will contain the contents of the message the action was initiated from.
```

Comment:
"the content object will contain the contents of the message" -- I think would benefit from an example of the payload. As a developer I'd really want to know what bits of the message I'm getting, and what that looks like.
- Do I get the message content as text, or do I get the raw HTML as well?
- Do I get all the entities in the message? mentions, attachments, etc?
- Do I get the message id? the conversation id of the thread the message is in?

Reply:
I've added an example of the request sent to the bot that should address this.